easybuild/easyconfigs/p/PyTorch/PyTorch-1.11.0-foss-2021b-CUDA-11.4.1.eb

```diff
@@ -0,0 +1,126 @@
+    'PyTorch-1.9.0_limit-world-size-for-zero-redundancy-opt-test.patch',
+    'PyTorch-1.10.0_fix-test-dataloader-fixed-affinity.patch',
+    'PyTorch-1.10.0_skip_cmake_rpath.patch',
+    'PyTorch-1.11.0_increase-distributed-test-timeout.patch',
+    'PyTorch-1.11.0_skip_failing_ops_tests.patch',
+    'PyTorch-1.11.0_fix_skip_jit_cuda_fuser.patch',
+    'PyTorch-1.11.0_fix_sharded_imports.patch',
+    'PyTorch-1.11.0_increase_test_tolerances_TF32.patch',
+    'PyTorch-1.11.0_increase_c10d_gloo_timeout.patch',
+    'PyTorch-1.11.0_disable_failing_jit_cuda_fuser_tests.patch',
+]
+checksums = [
+    None,  # can't add proper SHA256 checksum, because source tarball is created locally after recursive 'git clone'
+    'b899aa94d9e60f11ee75a706563312ccefa9cf432756c470caa8e623991c8f18',  # PyTorch-1.7.0_avoid-nan-in-test-torch.patch
+    '622cb1eaeadc06e13128a862d9946bcc1f1edd3d02b259c56a9aecc4d5406b8a',  # PyTorch-1.7.0_disable-dev-shm-test.patch
+    '89ac7a8e9e7df2e64cf8404fe3a279f5e9b759fee41c9de3aaff9c22f385c2c6',  # PyTorch-1.8.1_dont-use-gpu-ccc-in-test.patch
+    # PyTorch-1.9.0_limit-world-size-for-zero-redundancy-opt-test.patch
+    'ff573660913ce055e24cfd194ce747ba5685091c631cfd443eae2a99d56b57ea',
+    # PyTorch-1.10.0_fix-test-dataloader-fixed-affinity.patch
+    '313dca681f45ce3bc7c4557fdcdcbe0b77216d2c708fa30a2ec0e22c44876707',
+    'ac05943bb205623f91ef140aa00869efc5fe844184bd666bebf5405808610448',  # PyTorch-1.10.0_skip_cmake_rpath.patch
+    # PyTorch-1.11.0_increase-distributed-test-timeout.patch
+    '087ad20163a1291773ae3457569b80523080eb3731e210946459b2333a919f3f',
+    '8eaca92d64fcadb0552d28e9c7ea5c4bc669d2fe33004e45a3519ce8d0d136a2',  # PyTorch-1.11.0_skip_failing_ops_tests.patch
+    '21fc678febcdfbb9dabd72235be23cd392044e9a954f6580d15b530e1f69dcc1',  # PyTorch-1.11.0_fix_skip_jit_cuda_fuser.patch
+    '9a04f4285b800dad8a00c3014af0a9713d40d5dd35d10931c7c0da4e89c558e9',  # PyTorch-1.11.0_fix_sharded_imports.patch
+    # PyTorch-1.11.0_increase_test_tolerances_TF32.patch
+    '26e179a4f6f57e49209092612ae5f5cd8c03fd2ca84566ba0244eabefc3736ba',
+    # PyTorch-1.11.0_increase_c10d_gloo_timeout.patch
+    '20cd4a8663f74ab326fdb032b926bf5c7e94d9750c515ab9050927ba00cf1953',
+    # PyTorch-1.11.0_disable_failing_jit_cuda_fuser_tests.patch
+    'e7bfe120a8b3fe2b40dac6839852a5fbab3cb3429fbe44a0fc3a1800adaaee51',
+]
+
+osdependencies = [OS_PKG_IBVERBS_DEV]
+
+builddependencies = [
+    ('CMake', '3.22.1'),
+    ('hypothesis', '6.14.6'),
+]
+
+dependencies = [
+    ('CUDA', '11.4.1', '', True),
+    ('Ninja', '1.10.2'),  # Required for JIT compilation of C++ extensions
+    ('Python', '3.9.6'),
+    ('protobuf', '3.17.3'),
+    ('protobuf-python', '3.17.3'),
+    ('pybind11', '2.7.1'),
+    ('SciPy-bundle', '2021.10'),
+    ('typing-extensions', '4.3.0'),
+    ('PyYAML', '5.4.1'),
+    ('MPFR', '4.1.0'),
+    ('GMP', '6.2.1'),
+    ('numactl', '2.0.14'),
+    ('FFmpeg', '4.3.2'),
+    ('Pillow', '8.2.0'), # 8.3.2
```

Comment:
Note that Pillow 8.3.2 is used in this toolchain as dependency. I see you put that in a comment, was there a particular reason not to use it here?

Reply:
I initially tried with Pillow 8.3.2, but it has a compatibility error. The configuration requires Pillow lower version. Hence I used Pillow 8.2.0 as the same version was used in PyTorch-1.11.0-foss-2021a-CUDA-11.3.1.eb

Reply:
Ok, in that case two things will need to be done:

1. A clear comment should be put in the EasyConfig as to why the version is different. Typically a one-sentence mention of the error you'd get if you use newer pillow, plus a reference link to an issue describing the error more extensively (if needed). Feel free to just open an issue for that on `easybuild-easyconfigs`, describe the error with a full trace of the output/error you get, and close your own issue with a remark that the solution is to downgrade to Pillow 8.2.0. 

2. An exception needs to be made for the CI to accept that a different version of Pillow will be used here than in other parts of the toolchain. You can do that by adding `Pillow` to the dictionary [here](https://github.com/easybuilders/easybuild-easyconfigs/blob/fdcaa750ca492ff2bb5cc122b51806767b9a4128/test/easyconfigs/easyconfigs.py#L488). The key should be `Pillow` and the value should be a list of tuples, where the first element is the version of `pillow` that should be accepted (i.e. `'8.2.0'` in this case) and the second should be a list of the packages that are allowed to use this dependency version of `Pillow`, i.e. in your case `r'PyTorch-1\.11\.0-`. These are regular expressions, so that's why the `.` should be escaped there (there are plenty of examples in that file).

By the way, I see the CI is also tripping over the `typing-extensions` version, for the same reason (another version of `typing-extensions` is already in use in this toolchain). The same two points as above will apply to that. I.e. justify & document why we deviate from the version already use in the toolchain (1) and add it as an exception to CI (2).